### ISSUE TITLE
The different model versions available in TerraTorch

### ISSUE THREAD

**Issue by ArthurDelannoyazerty** [IBM/terramind#5]
Hello,

Can you explain the different models and backbones availables in Terratorch ?
Backbones availables (BACKBONE_REGISTRY) :
- terramind_v1_base
- terramind_v1_base_tim
- terramind_v1_large
- terramind_v1_large_tim

The differences between `base` and `large` is the size of the model. But what exactly is different between `base` and `base_tim` ?

FULL_MODEL_REGISTRY contains the tokenizer for the differents modalities, but there is no tokenizer available for the `caption` and `coords` modality, how can I tokenize the text modalites ?

**Comment by blumenstiel** [comment]
The `_tim` models use the Thinking-in-Modalities (TiM) approach which is described in the paper in more details. So you can define `tim_modalities` as intermediate modalities.

About the caption and coords tokenizer, that is still an todo. It's a adapted Wordpiece tokenizer. I have to see when I have the time to add it to TerraTorch.

**Comment by ArthurDelannoyazerty** [comment]
Hi,
ok I understand the TIM working now : If we input modalities and some are missing, we generate the missing ones (except the target ones) and the geenrated the target modality.

Thank you for your response

**Comment by blumenstiel** [comment]
Exactly. Just to clarify: With target modality, we do not mean one of the pre-training ones but the actual targets in fine-tuning, e.g., a segmentation mask.
